pkg/networking/cilium/upgrader.go

```diff
@@ -138,11 +139,15 @@ func ChangeDiff(currentSpec, newSpec *cluster.Spec) *types.ChangeDiff {
 }
 
 func ciliumHelmChartValuesChanged(currentSpec, newSpec *cluster.Spec) bool {
-	if currentSpec.Cluster.Spec.ClusterNetwork.CNIConfig != nil {
-		if currentSpec.Cluster.Spec.ClusterNetwork.CNIConfig.Cilium != nil {
```

Comment:
Are you missing this check now?

Reply:
not checking on purpose because it won't be needed. For clusters created with release <= 0.7, CNIConfig != nil check will be enough. For clusters with newer versions, by this point in the cilium upgrader logic, CiliumConfig will be initialized so we don't need to check